browser/ui/webui/settings/brave_email_aliases_handler.cc

```diff
@@ -0,0 +1,447 @@
+}
+
+void BraveEmailAliasesHandler::RequestAccount(const base::Value::List& args) {
+  CHECK_EQ(2U, args.size());
+  const auto callback_id = args[0].GetString();
+  const auto account_email = args[1].GetString();
+  AllowJavascript();
+  const auto bodyValue = base::Value::Dict()
+                             .Set("email", account_email)
+                             .Set("intent", "auth_token")
+                             .Set("service", "email-aliases");
+  ApiFetch(
+      GURL(kAccountsServiceRequestURL), net::HttpRequestHeaders::kPostMethod,
+      std::nullopt, bodyValue,
+      base::BindOnce(&BraveEmailAliasesHandler::OnRequestAccountResponse,
+                     weak_factory_.GetWeakPtr(), callback_id, account_email));
+}
+
+void BraveEmailAliasesHandler::OnRequestAccountResponse(
+    const std::string& callback_id,
+    const std::string& account_email,
+    std::optional<std::string> response_body) {
+  AllowJavascript();
+  if (response_body) {
+    std::optional<base::Value> response_value =
+        base::JSONReader::Read(response_body.value());
+    if (response_value && response_value->is_dict()) {
+      const auto* verification_token =
+          response_value->GetDict().Find("verificationToken");
+      if (verification_token && verification_token->is_string()) {
+        // Store the verification token while we wait for session confirmation.
+        SetStringPref(kEmailAliasesAccountEmail, account_email);
+        SetStringPref(kEmailAliasesVerificationToken,
+                      verification_token->GetString());
+        // Acknowledge success to the caller.
+        ResolveJavascriptCallback(base::Value(callback_id), base::Value());
+        return;
+      }
+    }
+  }
+  RejectJavascriptCallback(base::Value(callback_id),
+                           base::Value("no verification token"));
+}
+
+void BraveEmailAliasesHandler::GetAccountEmail(const base::Value::List& args) {
+  CHECK_EQ(1U, args.size());
+  AllowJavascript();
+  const auto callback_id = args[0].GetString();
+  const auto account_email = GetStringPref(kEmailAliasesAccountEmail);
+  ResolveJavascriptCallback(base::Value(callback_id),
+                            base::Value(account_email));
+}
+
+void BraveEmailAliasesHandler::Logout(const base::Value::List& args) {
+  AllowJavascript();
+  ClearPref(kEmailAliasesAccountEmail);
+  ClearPref(kEmailAliasesVerificationToken);
+  ClearPref(kEmailAliasesAuthToken);
+  ResolveJavascriptCallback(base::Value(args[0].GetString()), base::Value());
+}
+
+void BraveEmailAliasesHandler::CloseBubble(const base::Value::List& args) {
+  CHECK_EQ(1U, args.size());
+  email_aliases::EmailAliasesBubbleUI::Close();
+}
+
+void BraveEmailAliasesHandler::FillField(const base::Value::List& args) {
+  CHECK_EQ(2U, args.size());
+  AllowJavascript();
+  const auto callback_id = args[0].GetString();
+  const auto field_value = args[1].GetString();
+  email_aliases::EmailAliasesBubbleUI::FillField(field_value);
```

Comment:
these static methods/variables are a major problem here. What exactly are you trying to do?